src/metadata/api.ts

```diff
@@ -125,11 +85,6 @@ export class MetadataApi {
    * @param appData
    */
   async calculateAppDataHash(appData: AnyAppDataDocVersion): Promise<IpfsHashInfo | void> {
-    const validation = await this.validateAppDataDoc(appData)
-    if (!validation?.success) {
-      throw new CowError('Invalid appData provided', validation?.errors)
-    }
-
```

Comment:
I don't think we should remove the validation

Reply:
Discussed, moved back and added lazy-loading